File: apps/junit5-example/src/test/kotlin/example/GherkinPatternsStoryTest.kt
package example

import dev.executablestories.junit5.Story
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.assertTrue

class GherkinPatternsStoryTest {

    @Test
    fun multipleGivenAutoAnd() {
        Story.init("User logs in successfully")
        Story.given("the user account exists")
        Story.given("the user is on the login page")
        Story.given("the account is active")
        Story.`when`("the user submits valid credentials")
        Story.then("the user should see the dashboard")
        assertTrue(true)
    }

    @Test
    fun multipleWhenAutoAnd() {
        Story.init("User updates profile settings")
        Story.given("the user is logged in")
        Story.`when`("the user navigates to settings")
        Story.`when`("the user changes their display name")
        Story.then("the changes should be saved")
        assertTrue(true)
    }

    @Test
    fun multipleThenAutoAnd() {
        Story.init("Successful order confirmation")
        Story.given("the user has items in cart")
        Story.`when`("the user completes checkout")
        Story.then("the order should be created")
        Story.then("a confirmation email should be sent")
        Story.then("the inventory should be updated")
        assertTrue(true)
    }

    @Test
    fun butKeywordForContrast() {
        Story.init("Login blocked for suspended user")
        Story.given("the user account exists")
        Story.given("the account is suspended")
        Story.`when`("the user submits valid credentials")
        Story.then("the user should see an error message")
        Story.but("the user should not be logged in")
        Story.but("the session should not be created")
        assertTrue(true)
    }

    @Test
    fun explicitAndSteps() {
        Story.init("Order with explicit And steps")
        Story.given("the user is logged in")
        Story.and("the user has a valid payment method")
        Story.and("the user has items in cart")
        Story.`when`("the user clicks checkout")
        Story.and("confirms the order")
        Story.then("the order should be created")
        Story.and("the payment should be processed")
        assertTrue(true)
    }
}

File: apps/xunit-example/GherkinPatternsStoryTest.cs
using ExecutableStories.Xunit;
using Xunit;

namespace XunitExample.Tests;

public class GherkinPatternsStoryTest
{
    [Fact]
    public void Multiple_Given_auto_And()
    {
        Story.Init("User logs in successfully");
        Story.Given("the user account exists");
        Story.Given("the user is on the login page");
        Story.Given("the account is active");
        Story.When("the user submits valid credentials");
        Story.Then("the user should see the dashboard");
        Story.RecordAndClear();
    }

    [Fact]
    public void Multiple_When_auto_And()
    {
        Story.Init("User updates profile settings");
        Story.Given("the user is logged in");
        Story.When("the user navigates to settings");
        Story.When("the user changes their display name");
        Story.Then("the changes should be saved");
        Story.RecordAndClear();
    }

    [Fact]
    public void Multiple_Then_auto_And()
    {
        Story.Init("Successful order confirmation");
        Story.Given("the user has items in cart");
        Story.When("the user completes checkout");
        Story.Then("the order should be created");
        Story.Then("a confirmation email should be sent");
        Story.Then("the inventory should be updated");
        Story.RecordAndClear();
    }

    [Fact]
    public void But_keyword_for_contrast()
    {
        Story.Init("Login blocked for suspended user");
        Story.Given("the user account exists");
        Story.Given("the account is suspended");
        Story.When("the user submits valid credentials");
        Story.Then("the user should see an error message");
        Story.But("the user should not be logged in");
        Story.But("the session should not be created");
        Story.RecordAndClear();
    }

    [Fact]
    public void Explicit_And_steps()
    {
        Story.Init("Order with explicit And steps");
        Story.Given("the user is logged in");
        Story.And("the user has a valid payment method");
        Story.And("the user has items in cart");
        Story.When("the user clicks checkout");
        Story.And("confirms the order");
        Story.Then("the order should be created");
        Story.And("the payment should be processed");
        Story.RecordAndClear();
    }
}
